packages/components/src/text-control-with-affixes/style.scss

```diff
@@ -0,0 +1,135 @@
+.text-control-with-affixes {
+	display: inline-flex;
+	flex-direction: column;
+    width: 100%;
+
+    :nth-child(1) {
+        border-top-left-radius: 4px;
+        border-top-right-radius: 4px;
+    }
+
+    :nth-last-child(1) {
+        border-bottom-left-radius: 4px;
+        border-bottom-right-radius: 4px;
+    }
+    
+    &.no-wrap {
+        flex-direction: row;
+        
+        :nth-child(1) {
+            border-top-right-radius: 0;
+            border-top-left-radius: 4px;
+            border-bottom-left-radius: 4px;
+        }
+    
+        :nth-last-child(1) {
+            border-bottom-left-radius: 0;
+            border-top-right-radius: 4px;
+            border-bottom-right-radius: 4px;
+        }
+    }
+
+    @include breakpoint( ">400px" ) {
+        flex-direction: row;
+        
+        :nth-child(1) {
+            border-top-right-radius: 0;
+            border-top-left-radius: 4px;
+            border-bottom-left-radius: 4px;
+        }
+    
+        :nth-last-child(1) {
+            border-bottom-left-radius: 0;
+            border-top-right-radius: 4px;
+            border-bottom-right-radius: 4px;
+        }
+    }
+
+    input[type="email"],
+    input[type="password"],
+    input[type="url"],
+    input[type="text"],
+    input[type="number"] {
+        flex-grow: 1;
+        margin: 0;
+
+        &:focus {
+			// Fixes the right border of the box shadow displayed when this input element is focused which appears as
+			// cut off when this input has a suffix, or is stuck to another element that has a higher stacking order
+			// (fix found at http://stackoverflow.com/a/24728957)
+			transform: scale( 1 );
+		}
+
+		&:disabled {
+			border-right-width: 0;
+
+			& + .text-control-with-affixes__suffix {
+				border-left: 1px solid $core-grey-light-500;
+			}
+		}
+    }
+}
+
+@mixin no-prefix-wrap() {
+	border-right: none;
+}
+
+@mixin no-suffix-wrap() {
+	border-left: none;
+}
+
+.text-control-with-affixes__prefix,
+.text-control-with-affixes__suffix {
+	position: relative;
+	background: $core-grey-light-100;
+	border: 1px solid $core-grey-light-500;
+	color: $gray-text;
+	padding: 7px 14px;
+	white-space: nowrap;
+	flex: 1 0 auto;
+	font-size: 14px;
+	line-height: 1.5;
+}
+
+.text-control-with-affixes__prefix {
```

Comment:
Can you prevent this from wrapping to two lines when the viewport is less than 400px please?

Reply:
just clarifying that last comment... 

 At the smallest viewport sizes, I noticed the entry field wraps under the the prefix, and decreases in height. 

<img width="372" alt="screen shot 2018-12-12 at 8 17 08 pm" src="https://user-images.githubusercontent.com/4500952/49915616-53fea600-fe4b-11e8-9265-240143587c6c.png">

Can you keep it all on the same line, even at the smallest sizes, please?

Reply:
Do we never want it to wrap? I can remove the `noWrap` stuff (just ported from Calypso) as well then.

Reply:
Yeah, I'd just prefer it to never wrap, so that'd be great. Thanks!

Reply:
Done in e3a45da.